R/msaGaugeRR.R

```diff
@@ -532,90 +532,81 @@ msaGaugeRR <- function(jaspResults, dataset, options, ...){
   return(plot)
 }
 
-.gaugeByPartGraph <- function(dataset, measurements, parts, operators, options){
-
-  dataset <- tidyr::gather(dataset, Repetition, Measurement, measurements[1]:measurements[length(measurements)], factor_key=TRUE)
-
-  means <- aggregate(dataset["Measurement"], dataset[parts], mean)
-
+.gaugeByPartGraph <- function(dataset, measurements, parts, operators, options, ready){
 
   plot <- createJaspPlot(title = gettext("Measurements by Part"))
 
   plot$dependOn(c("gaugeByPart", "gaugeByPartAll", "gaugeRRmethod"))
 
-  p <- ggplot2::ggplot()
+  if (ready){
+    dataset <- tidyr::gather(dataset, Repetition, Measurement, measurements[1]:measurements[length(measurements)], factor_key=TRUE)
+    means <- aggregate(dataset["Measurement"], dataset[parts], mean)
+
+    p <- ggplot2::ggplot()
 
-  if(options$gaugeByPartAll)
-    p <- p + jaspGraphs::geom_point(data = dataset, ggplot2::aes_string(x = parts, y = "Measurement"), col = "gray")
+    if(options$gaugeByPartAll)
+      p <- p + jaspGraphs::geom_point(data = dataset, ggplot2::aes_string(x = parts, y = "Measurement"), col = "gray")
 
+    p <- p + jaspGraphs::geom_point(data = means, ggplot2::aes_string(x = parts, y = "Measurement")) +
```

Comment:
see the other comment about ggplot2 structure. Also, why `aes_string`?

Reply:
I have changed the ggplot2 structure. I can't make it work without aes_string, though. `parts` is a string, so using aes() gives not the output I want...